app/main/views/suppliers.py

```diff
@@ -409,25 +408,52 @@ def update_supplier_framework_details(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
+    # `agreementDetails` shouldn't be passed in unless the framework has framework_agreement_details
+    if 'agreementDetails' in update_json and framework.framework_agreement_details is None:
+        abort(400, "Framework '{}' does not accept 'agreementDetails'".format(framework_slug))
+
+    if (
+            (framework.framework_agreement_details and framework.framework_agreement_details.get('frameworkAgreementVersion')) and  # noqa
+            ('agreementDetails' in update_json or update_json.get('agreementReturned'))
+    ):
+        if update_json.get('agreementDetails') and update_json['agreementDetails'].get('uploaderUserId'):
```

Comment:
We decided to check this before doing anything else because we only want to check for users when an id is actually passed in.  We wanted to avoid checking a user exists every time an update is submitted.

We _could_ move all of this code to line 442 (so that we know anything in the `upload_data` would have already been validated) but we would still have to check for the existence of an `uploaderUserId` in the passed-in data.